dma/sniff_crc/sniff_crc.c

```
@@ -0,0 +1,88 @@
+    uint32_t crc_res;
+
+    stdio_init_all();
+
+    printf("\n");
+    printf("\n");
+
+    // cacluate and append the crc
+    crc_res = soft_crc32_block(CRC32_INIT, src, DATA_TO_CHECK_LEN);
+    *((uint32_t *)&src[DATA_TO_CHECK_LEN]) = crc_res;
+
+    printf("Buffer to DMA: ");
+    for (int i = 0; i < 12; i++) {
+        printf("0x%02x ", src[i]);
+    }
+    printf("\n");
+
+    // Get a free channel, panic() if there are none
+    int chan = dma_claim_unused_channel(true);
+
+    // 8 bit transfers. Both read and write address increment after each
+    // transfer (each pointing to a location in src or dst respectively).
+    // No DREQ is selected, so the DMA transfers as fast as it can.
+
+    dma_channel_config c = dma_channel_get_default_config(chan);
+    channel_config_set_transfer_data_size(&c, DMA_SIZE_8);
+    channel_config_set_read_increment(&c, true);
+    channel_config_set_write_increment(&c, true);
+
+    // Sniff specific set-up
+    channel_config_set_sniff_enable(&c, true);
+    dma_sniffer_set_data_accumulator(CRC32_INIT);
+    dma_sniffer_set_output_reverse_enabled(true);
+    dma_sniffer_enable(chan, DMA_SNIFF_CTRL_CALC_VALUE_CRC32R, true);
+
+    dma_channel_configure(
+        chan,          // Channel to be configured
+        &c,            // The configuration we just created
+        dst,           // The initial write address
+        src,           // The initial read address
+        TOTAL_LEN,     // Total number of transfers inc. appended CRC; each is 1 byte
+        true           // Start immediately.
+    );
+
+    // We could choose to go and do something else whilst the DMA is doing its
+    // thing. In this case the processor has nothing else to do, so we just
+    // wait for the DMA to finish.
+    dma_channel_wait_for_finish_blocking(chan);
+
+    uint32_t sniffed_crc = dma_sniffer_get_data_accumulator();
+    printf("Completed DMA copy of %d byte buffer, sniffed crc: 0x%08lx\n", TOTAL_LEN, sniffed_crc);
```

Maybe print success/fail? And or panic on failure rather than just printf?

---

Some comments, but looks useful to me. You need to add to the readme as well.